File: md5/md5.java
import java.math.BigInteger;
import java.security.MessageDigest;

public class md5 {
    // ATOMIC_FUNCTION:  md5(String s)
    // TAGS: string
    // LANG:  java
    // FUNCTION: md5(String s)
    public String  md5(String x){
        try{
            String plaintext = "your text here";
            MessageDigest m = MessageDigest.getInstance("MD5");
            m.reset();
            m.update(plaintext.getBytes());
            byte[] digest = m.digest();
            BigInteger bigInt = new BigInteger(1,digest);
            String hashtext = bigInt.toString(16);
            return hashtext;
        }catch(Exception e) {
            // do proper exception handling
            return null;
        }


    }
    public static void main(String args[]) {
        String start_val = "apple";
        String expected_end_val ="3b777055f53f39be20ba812d5cec7621";
        md5 m = new md5();

        String end_val = m.md5(start_val);


        // TEST
        if(end_val.equals( expected_end_val) ){
            System.out.println("PASS");
        } else {

            System.out.println("FAIL");
        }
    }

}

File: md5/md5.js
// ATOMIC FUNCTION:abs
// TAGS: math，numeric
// LANG: node.js
// FUNCTION: Math.abs(val, precision)
var start_val = "apple";
var crypto = require('crypto');
var expected_end_val="1f3870be274f6c49b3e31a0c6728957f";
var end_val=crypto.createHash('md5').update(start_val).digest("hex");

// TEST
if(end_val==expected_end_val){
    console.log('PASS');
}else{
    console.log('FAIL');
}
